qiskit/transpiler/passes/mapping/apply_layout.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+A pass for transforming a circuit with virtual qubits into a circuit with physical qubits.
+"""
+from qiskit.circuit import QuantumRegister
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.transpiler.basepasses import TransformationPass
+
+
+class ApplyLayout(TransformationPass):
+    """
+    Transform a DAGCircuit with virtual qubits into a DAGCircuit with physical qubits.
+    """
+
+    def __init__(self, coupling, initial_layout=None):
+        """
+        Transform a DAGCircuit with virtual qubits into a DAGCircuit with physical qubits
+        defined in the `coupling` by applying a given `initial_layout`.
+        Args:
+            coupling (CouplingMap): coupling graph to which the circuit is mapped
+            initial_layout (Layout): initial layout to be applied
+        """
+        super().__init__()
+        self._coupling = coupling
+        self._initial_layout = initial_layout
+
+    def run(self, dag):
+        """
+        Runs the ApplyLayout pass on `dag`.
+        Args:
+            dag (DAGCircuit): DAG to map.
+        Returns:
+            DAGCircuit: A mapped DAG (with physical qubits).
+        """
+        if not self._initial_layout:
+            self._initial_layout = self.property_set["layout"]
+
+        q = QuantumRegister(self._coupling.size(), 'q')
```

Comment:
The way it is written, this pass would change the number of qubits in the DAG. I don't think it should do that. It should just update the operands of each DAG Node to work on the updated physical qubit.

I don't think the pass needs to even take `coupling_map` as input. Applying a layout should just be a matter of taking a circuit + a layout, and relabeling the circuit's qubits based on that layout. 

The layout already contains information about which physical qubits must be used. Coupling map should just be used in the swap stage, not here.